README.md

```diff
@@ -1,2 +1,101 @@
 # patch-filters
 special project to host GSOC project Patch Suppressions 
+
+## PatchFilter Report Setup
+
+### Requirements
+
+- Checkstyle repository which including patch-filter need to be cloned.
+- [Contribution repository](https://github.com/checkstyle/contribution) need to be cloned.
+
+### Clone
+
+Just clone this repository to your local.
+
+```bash
+$ git clone git@github.com:checkstyle/patch-filters.git
+```
+
+### local install
+
+Because patch-filter has not been merged by Checkstyle, so you should local install this repository——patch-filters
+
+```bash
+$ cd /path/to/patch-filters
+$ mvn clean install
+```
+
+then add this dependency to checkstyle
+
+```xml
+<dependency>
+  <groupId>com.github.checkstyle</groupId>
+  <artifactId>patch-filter</artifactId>
+  <version>1.0-SNAPSHOT</version>
+</dependency>
+```
+then create a new branch, for example `patch-filter`
+
+```bash
+$ cd /path/to/checkstyle
+$ git checkout -b patch-filter
+$ git add .
+$ git commit -m "add patch-filter dependency"
+```
+### set checkstyle-tester
+
+Firstly, you need to do is to modify projects-to-test-on.properties and test project that you want.
+For example, if I want to test `guava`, then I will only remove # before `guava` (anything that starts with # 
+is considered a comment and is ignored.)
+
+Secondly, you need to set baseConfig.xml and patchConfig.xml by yourself, 
+**Attention:**
+ 
+ baseConfig does not need any checks in it and set `severity` to `ignore`, 
+like `<property name="severity" value="ignore"/>`; 
+
+patchConfig need Checks and patch-filter, added checks can
+refer to [here](https://github.com/checkstyle/checkstyle/blob/ec4d06712ab203d31d73c5c6d5c46067f3a6d5b3/config/checkstyle_checks.xml#L60-L190), 
+you can skip Translation Check and checks that reuqires extra config file (Header and RegExpHeader), patchfilter setting is like:
+```xml
+<module name="com.github.checkstyle.patchfilter.SuppressionPatchFilter">
+    <property name="file" value="path/to/patch-filters/DiffReport/patch.txt"/>
+</module>
+```
+value should be absolute path
+
+### Open this repo in IDEA or other IDE
+
+You can find `GeneratePatchFileLauncher` in `/path/to/patch-filters/src/main/java/com/github/checkstyle/generatepatchfile/`,
+then add seven follow program arguments in Run/Debug Configurations:
+
+```bash
+# RepoPath of repository that will be checked by checkstyle, here example is guava
+path/to/guava/
+# Path of checkstyle
+path/to/checkstyle/
+# Path of checkstyle-tester
+patch/to/checkstyle-contribution/checkstyle-tester/
+# Patch-branch name  of checkstyle, which should add patch-filters dependency as mentioned above, 
+# here is branch name for example patch-filter 
+patch-filter
```

Comment:
This is not required, after checkout to branch in contribution repo.

Reply:
yes, but as mentioned in https://github.com/checkstyle/patch-filters/pull/23#discussion_r443286882, generate() need this information to run `diff.groovy`

Reply:
checkstyle should stay in master branch, but checkstyle-tester should be switched to patch-filter-jar branch.

Reply:
done